lib/signaling/v2/peerconnection.js

```diff
@@ -488,10 +502,46 @@ class PeerConnectionV2 extends StateMachine {
       this._log.info('ICE reconnected');
     }
 
+    if (iceConnectionState === 'connected') {
+      this._mediaMonitor.start(() => {
+        this._log.debug(`makarand: Media Monitor detected inactivity while in "${this._lastIceConnectionState}" state`);
+        if (this._lastIceConnectionState === 'disconnected') {
+          this._mediaMonitor.stop();
+          if (!this._shouldRestartIce && !this._isRestartingIce) {
+            this._shouldRestartIce = true;
+            this.offer();
+          }
+        }
+      });
+    } else if (iceConnectionState === 'failed' || iceConnectionState === 'closed') {
+      this._mediaMonitor.stop();
+    }
+
     this._lastIceConnectionState = iceConnectionState;
     this.emit('iceConnectionStateChanged');
   }
 
+  // TODO: simplify and move to IceMediaMonitor once timestamp is added to the webrtc getStat api.
+  getMediaStats() {
```

Comment:
@makarandp0 This method won't be necessary due to availability of `IceConnectionMonitor#_getIceConnectionStats()`.